topostats/tracing/skeletonize.py

```diff
@@ -2,78 +2,641 @@
+            return self._skeletonize_medial_axis(self.mask)
+        if method == "thin":
+            return self._skeletonize_thin(self.mask)
+        if method == "joe":
+            return self._skeletonize_joe(self.image, self.mask)
+        raise ValueError(method)
+
+    @staticmethod
+    def _skeletonize_zhang(mask: np.ndarray) -> np.ndarray:
+        """Wrapper for the scikit image implimentation of the Zhang skeletonisation method.
+
+        Parameters
+        ----------
+        mask: np.ndarray
+            A binary array to skeletonise.
+
+        Returns
+        -------
+        np.ndarray
+            The mask array reduce to a single pixel thickness
+        """
+        return skeletonize(mask, method="zhang")
+
+    @staticmethod
+    def _skeletonize_lee(mask: np.ndarray) -> np.ndarray:
+        """Wrapper for the scikit image implimentation of the Lee skeletonisation method.
+
+        Parameters
+        ----------
+        mask: np.ndarray
+            A binary array to skeletonise.
+
+        Returns
+        -------
+        np.ndarray
+            The mask array reduce to a single pixel thickness
+        """
+        return skeletonize(mask, method="lee")
+
+    @staticmethod
+    def _skeletonize_medial_axis(image: np.ndarray) -> np.ndarray:
+        """Wrapper for the scikit image implimentation of the medial axis skeletonisation method.
+
+        Parameters
+        ----------
+        mask: np.ndarray
+            A binary array to skeletonise.
+
+        Returns
+        -------
+        np.ndarray
+            The mask array reduce to a single pixel thickness
+        """
+        # don't know how these work - do they need img or mask?
+        return medial_axis(image, return_distance=False)
+
+    @staticmethod
+    def _skeletonize_thin(image: np.ndarray) -> np.ndarray:
+        """Wrapper for the scikit image implimentation of the thin skeletonisation method.
+
+        Parameters
+        ----------
+        mask: np.ndarray
+            A binary array to skeletonise.
+
+        Returns
+        -------
+        np.ndarray
+            The mask array reduce to a single pixel thickness
+        """
+        # don't know how these work - do they need img or mask?
+        return thin(image)
+
+    @staticmethod
+    def _skeletonize_joe(image: np.ndarray, mask: np.ndarray) -> np.ndarray:
+        """Wrapper for Pyne-lab member Joe's skeletonisation method.
+
+        Parameters
+        ----------
+        mask: np.ndarray
+            A binary array to skeletonise.
+
+        Returns
+        -------
+        np.ndarray
+            The mask array reduce to a single pixel thickness
+
+        Notes
+        -----
+        This method is based on Zhang's method but produces different results
+        (less branches but slightly less accurate).
+        """
+        # image height bits don't seem to be used but are there?
+        return joeSkeleonize(image, mask).doSkeletonising()
```

Comment:
Typo here and in the class definition...

```suggestion
        return joeSkeletonize(image, mask).doSkeletonising()
```